kpatch-build/create-diff-object.c

```diff
@@ -608,16 +665,40 @@ static void kpatch_compare_sections(struct list_head *seclist)
 
 	/* sync symbol status */
 	list_for_each_entry(sec, seclist, list) {
-		if (is_rela_section(sec)) {
-			if (sec->base->sym && sec->base->sym->status != CHANGED)
-				sec->base->sym->status = sec->status;
-		} else {
-			if (sec->sym && sec->sym->status != CHANGED)
-				sec->sym->status = sec->status;
+		struct symbol *sym;
+		if (is_rela_section(sec))
+			sym = sec->base->sym;
+		else
+			sym = sec->sym;
+
+		if (sym && sym->status != CHANGED) {
+			sym->status = sec->status;
+			if (sym->parent) {
+				if (sec->status == NEW && sym->parent->sec->twin)
+					sym->parent->status = CHANGED;
```

Comment:
Since we decided that a new subfunction will always change the parent, I think this isn't needed?

Reply:
I think we better leave it here, otherwise we are dependent on the section ordering in the list. I.e. if the patch results in a NEW subfunction and parent section is already marked as CHANGED without this if we will be marking parent as NEW. Or am I missing something?

Reply:
It seems a little convoluted, what do you think about changing it to something like this?
```
	/* sync symbol status */
	list_for_each_entry(sec, seclist, list) {
		if (is_rela_section(sec)) {
			if (sec->base->sym && sec->base->sym->status != CHANGED)
				sec->base->sym->status = sec->status;
		} else {
			struct symbol *sym = sec->sym;

			if (sym && sym->status != CHANGED)
				sym->status = sec->status;

			if (sym && sym->child && sym->status == SAME &&
			    sym->child->sec->status == CHANGED)
				sym->status = CHANGED;
		}
	}
```

Reply:
That should work and is easier to read. Will use that, thanks.